datatypes/src/primitives/datetime.rs

```diff
@@ -0,0 +1,668 @@
+        )
+    }
+
+    fn _from_fixed_offset(offset: chrono::FixedOffset) -> Self {
+        let offset_seconds = offset.local_minus_utc();
+
+        if offset_seconds == 0 {
+            return UTC;
+        }
+
+        let minute_in_secs = 60;
+        let hour_in_secs = 3600;
+
+        let hours = (offset.local_minus_utc() / hour_in_secs) as i8;
+        let minutes = (offset.local_minus_utc() % hour_in_secs / minute_in_secs) as i8;
+
+        Self { hours, minutes }
+    }
+}
+
+impl From<DateTime> for TimeInstance {
+    fn from(datetime: DateTime) -> Self {
+        Self::from(&datetime)
+    }
+}
+
+impl From<&DateTime> for TimeInstance {
+    fn from(datetime: &DateTime) -> Self {
+        TimeInstance::from_millis_unchecked(datetime.datetime.timestamp_millis())
+    }
+}
+
+impl TryFrom<TimeInstance> for DateTime {
+    type Error = DateTimeError;
+
+    fn try_from(time_instance: TimeInstance) -> Result<Self, Self::Error> {
+        use chrono::TimeZone;
+
+        if time_instance < TimeInstance::MIN || time_instance > TimeInstance::MAX {
+            return Err(DateTimeError::OutOfBounds);
+        }
+
+        match chrono::Utc.timestamp_millis_opt(time_instance.inner()) {
+            chrono::LocalResult::Single(datetime) => Ok(Self {
+                datetime: datetime.naive_utc(),
+                offset: UTC,
+            }),
+            chrono::LocalResult::None | chrono::LocalResult::Ambiguous(_, _) => {
+                Err(DateTimeError::OutOfBounds)
+            }
+        }
+    }
+}
+
+impl From<chrono::DateTime<chrono::FixedOffset>> for DateTime {
+    fn from(datetime: chrono::DateTime<chrono::FixedOffset>) -> Self {
+        let offset_secs = datetime.offset().fix().local_minus_utc();
+
+        if offset_secs == 0 {
+            return Self {
+                datetime: datetime.naive_utc(),
+                offset: UTC,
+            };
+        }
+
+        let hours = offset_secs / 3600;
+        let minutes = (offset_secs - hours * 3600) / 60;
+
+        Self {
+            datetime: datetime.naive_utc(),
+            offset: UtcOffset {
+                hours: hours as i8,
+                minutes: minutes as i8,
+            },
+        }
+    }
+}
+
+impl From<chrono::DateTime<chrono::Utc>> for DateTime {
+    fn from(datetime: chrono::DateTime<chrono::Utc>) -> Self {
+        Self {
+            datetime: datetime.naive_utc(),
+            offset: UTC,
+        }
+    }
+}
+
+impl From<DateTime> for chrono::DateTime<chrono::FixedOffset> {
+    fn from(datetime: DateTime) -> Self {
+        Self::from(&datetime)
+    }
+}
+
+impl From<&DateTime> for chrono::DateTime<chrono::FixedOffset> {
+    fn from(datetime: &DateTime) -> Self {
+        chrono::DateTime::from_utc(datetime.datetime, datetime.offset._to_fixed_offset())
+    }
+}
+
+pub enum DateTimeParseFormat {
+    Ymd,
+    Unix,
+    Custom {
+        fmt: String,
+        has_tz: bool,
+        has_time: bool,
+    },
+}
+
+impl DateTimeParseFormat {
```

Comment:
We need to adjust the datasets definitions and extend the documentation if we introduce a new custom parseformat

Reply:
Yes, unfortunately the old stuff works because %Y still gets passed to chrono and it works… Ideas for that?

Reply:
then we have to adjust all the dataset definitions?

Reply:
We could do it, but we first need to be sure how we want it. Is is okay this way? Will we allow modifiers? Do we need to escape the chrono stuff so that only our Placeholders are really used?

Reply:
idk, maybe we have to discuss it. Your format is certainly more verbose than what we had before

Reply:
[13762a7](https://github.com/geo-engine/geoengine/pull/523/commits/13762a70276d19b757a3de57880c813c2345afb8)